lib/user-interface/react/src/shared/model/model-management.model.ts

```diff
@@ -29,11 +31,142 @@ export enum ModelType {
     embedding = 'embedding',
 }
 
+export enum InferenceContainer {
+    TGI = 'tgi',
+    TEI = 'tei',
+    VLLM = 'vllm',
+    INSTRUCTOR = 'instructor',
+}
+
+export type IContainerHealthCheckConfig = {
+    Command: string[];
+    Interval: number;
+    StartPeriod: number;
+    Timeout: number;
+    Retries: number;
+};
+
+export type IContainerConfigImage = {
+    BaseImage: string;
+    Path: string;
+    Type: string;
+};
+
+export type IMetricConfig = {
+    AlbMetricName: string;
+    TargetValue: number;
+    Duration: number;
+    EstimatedInstanceWarmup: number;
+};
+
+export type ILoadBalancerHealthCheckConfig = {
+    Path: string;
+    Interval: number;
+    Timeout: number;
+    HealthyThresholdCount: number;
+    UnhealthyThresholdCount: number;
+};
+
+export type ILoadBalancerConfig = {
+    HealthCheckConfig: ILoadBalancerHealthCheckConfig
+};
+
+export type IAutoScalingConfig = {
+    MinCapacity: number;
+    MaxCapacity: number;
+    Cooldown: number;
+    DefaultInstanceWarmup: number;
+    MetricConfig: IMetricConfig;
+};
+
+export type IContainerConfig = {
+    BaseImage: IContainerConfigImage;
+    SharedMemorySize: number;
+    HealthCheckConfig: IContainerHealthCheckConfig;
+    Environment?: Record<string, string>[];
+};
+
 export type IModel = {
```

Comment:
I'll have to edit this in my PR, so don't worry about for now, but I'm also adding a "UniqueId" which is the UUID that comes from LiteLLM, which is how we'd eventually reference the model for modifications/mutations if we need to do anything from the LiteLLM perspective.